iceoryx_hoofs/legacy/include/iceoryx_hoofs/cxx/attributes.hpp

```diff
@@ -18,11 +18,12 @@
 #define IOX_HOOFS_CXX_ATTRIBUTES_HPP
 
 #include "iox/attributes.hpp"
+#include "iox/detail/deprecation_marker.hpp"
+
+IOX_DEPRECATED_HEADER_SINCE(3, "Please include 'iox/attributes.hpp' instead.")
 
 namespace iox
 {
-/// @todo iox-#1593 Deprecate include
-/// [[deprecated("Deprecated in 3.0, removed in 4.0, please include 'iox/attributes.hpp' instead")]]
```

Comment:
@elBoberido IIRC it was the namespace that got deprecated. So I think you need to move the new macro down to line 25. Same for all other headers.

Reply:
The problem with the namespace deprecation is that we need to port all classes from the respective namespace or we would get the warning also for those classes not yet ported. Additionally, it might also show the wrong file, e.g.
```cpp
#include "foo/bar.hpp" // deprecated
#include "foo/bas.hpp" // not yet deprecated
#include "foo/baz.hpp" // deprecated

Bar bar;
Bas bas; 
Baz baz; 
``` 
All, `bar`, `bas` and `baz` would show the deprecation warning set in `foo/baz.hpp`. Furthermore, if `foo/bas.hpp` would be the last include, no warnings would be shown at all.

I can add a todo-#1593 to add the namespace deprecation once all classes are ported

Reply:
Todo is added on a local branch